crates/red_knot_python_semantic/resources/mdtest/subscript/bytes.md

```diff
@@ -23,6 +23,18 @@ y = b[-6]  # error: [index-out-of-bounds] "Index -6 is out of bounds for bytes l
 reveal_type(y)  # revealed: Unknown
 ```
 
+## Slices
+
+```py
+b = b"\x00abc\xff"
+
+reveal_type(b[0:2])  # revealed: Literal[b"\x00a"]
+reveal_type(b[-3:])  # revealed: Literal[b"bc\xff"]
+
+a = b[::0]  # error: [slice-step-zero]
+reveal_type(a)  # revealed: Unknown
```

Comment:
This is a weird edge case that pyright and mypy do not detect (mypy [crashes](https://mypy-play.net/?mypy=master&python=3.13&gist=2e56de8c5fc30cf5e37fe15348c341bf)). I'm not sure if it's worth having a separate diagnostic for, but it was easily to implement. We could also ignore it and simplify infer a non-literal type (`slice`) for something like `::0`.